resources/fetching/fetchers/azure/assets_enricher_storage_account.go

```diff
@@ -0,0 +1,76 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fetchers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/samber/lo"
+
+	"github.com/elastic/cloudbeat/resources/fetching/cycle"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
+	"github.com/elastic/cloudbeat/resources/utils/strings"
+)
+
+type storageAccountEnricher struct {
+	provider azurelib.ProviderAPI
+}
+
+func (e storageAccountEnricher) Enrich(ctx context.Context, cycleMetadata cycle.Metadata, assets []inventory.AzureAsset) error {
+	subscriptions, err := e.provider.GetSubscriptions(ctx, cycleMetadata)
+	if err != nil {
+		return fmt.Errorf("storageAccountEnricher: error while getting subscription: %w", err)
+	}
+
+	diagSettings, err := e.provider.ListDiagnosticSettingsAssetTypes(ctx, cycleMetadata, lo.Keys(subscriptions))
```

Comment:
Why use `lo` and not [exp/maps](https://pkg.go.dev/golang.org/x/exp/maps#Keys)? 

PS: I still don't understand why this wasn't move from exp to [std](https://pkg.go.dev/maps) with go 1.21 😢

Reply:
Yes I agree `exp/maps` might be better but cloudbeat codebase uses `lo` so far, so perhaps we can open an issue to replace it cloudbeat-wide?

Reply:
Jah sure let's stick to lo if that's the pattern. Just curious why :)